Simperium/src/main/java/com/simperium/client/Channel.java

```diff
@@ -540,9 +539,10 @@ public void run() {
                 JSONObject extra = new JSONObject();
                 try {
                     extra.put("bucketName", mBucket.getName());
-                    extra.put("build", Version.BUILD);
-                    extra.put("version", Version.NUMBER);
-                    extra.put("client", Version.NAME);
+                    // TODO: Figure out what to do about these extra properties now that version is not available
```

Comment:
This is one of the TODOs I mentioned for removing the generated version information. I'd like to get some help on figuring out how to address this from @danilo04 & maybe @nsakaimbo as you'd have more information about its usage than I do.

To summarize, the version information is not available at the time of code compilation so it can not be used within the library. This project was using git hash for these values which we can still use. However, I'd much prefer if the build information is provided by the client as a passed value instead.

[It's worth noting that Android Gradle Plugin removed the version information for libraries from generated the `BuildConfig` class for similar reasons.](https://developer.android.com/studio/releases/gradle-plugin#version_properties_removed_from_buildconfig_class_in_library_projects)

Reply:
Based on what I found by asking the team, this is important information for debugging purposes. It is important to identify what client and version generated the different traces.

Reply:
That would mean some kind of hook in the gradle plugin that sits in between the version definition (reads git commit sha1) and the build. We could then write the `Version` file and then build.